src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlDeclaration.cs

```diff
@@ -60,10 +67,10 @@ public string Standalone
             }
         }
 
-        public override string Value
+        public override string? Value
```

Comment:
IIRC those are very messy and I ended up changing it back and forth couple of times: some implementation clearly return null and some other do not allow it so I ended up marking it as nullable. I'll take one more pass on this

Reply:
Those (Value, InnerText, Data) are absolutely inconsistent across inherited classes the more I look at it. Some return null, some don't expect null, some are accepting null on setter but don't return null, some throw for null on setter. I'm not sure how to annotate it correctly. Especially Value:
i.e. Value is always null for XmlEntityReference and XmlNode

I think regardless of what I do here it will not be correct.

Thoughts?

Reply:
Sounds like similar to Object.ToString() case, shouldn't we handle the same: make the parent nullable and inherited classes will have whatever nullability they could/should have